lib/lsperfm/core.rb

```diff
@@ -0,0 +1,81 @@
+require 'lsperfm/core/reporter'
+require 'lsperfm/core/run'
+
+module LogStash::PerfM
+
+  class ConfigException < Exception; end
+
+  class Core
+
+    attr_reader :definition, :install_path, :runner, :config
+
+    def initialize(definition, install_path, config='', runner = LogStash::PerfM::Runner)
+      @definition   = definition
+      @install_path = install_path
+      @runner       = runner
+      @config       = load_config(config)
+    end
+
+    def run(debug=false, headers=false)
+      tests    = load_tests(definition)
+      lines    = (headers ? ["name, #{runner.headers.join(',')}"] : [])
+      reporter = LogStash::PerfM::Reporter.new.start
+      tests.each do |test|
+        events  = test[:events].to_i
+        time    = test[:time].to_i
+
+        manager = runner.new(test_config(test[:config]), debug, install_path)
+        metrics = manager.run(events, time, runner.read_input_file(test_input(test[:input])))
+        lines << formatter(test[:name], metrics)
+      end
+      lines
+    rescue Errno::ENOENT => e
+      raise ConfigException.new(e)
+    ensure
+      reporter.stop if reporter
+    end
+
+    def config=(config)
+      @config = load_config(config)
+    end
+
+    private
+
+    def load_tests(definition)
+      return load_default_tests if definition.empty?
+      eval(IO.read(definition))
+    end
+
+    def load_default_tests
+      require 'lsperfm/defaults/suite.rb'
+      LogStash::PerfM::DEFAULT_SUITE
+    end
+
+    def load_config(config)
+      ::YAML::load_file(config)['default'] rescue {}
+    end
+
+    def test_config(file)
```

Comment:
I'm not a fan of calling these methods test_* because methods are usually actions, which means you're expecting test_config to test..a config. maybe .. get_config_path(file) or full_config_path(file)?